lib/auth.ts

```diff
@@ -482,9 +474,82 @@ const getAuth = function (
 			body: credentials,
 			sendToken: false,
 		});
+
 		return body;
 	}
 
+	/**
+	 * @summary Verifies an email
+	 * @name verifyEmail
+	 * @public
+	 * @function
+	 * @memberof balena.auth
+	 *
+	 * @param {Object} verificationPayload - in the form of email, and token
+	 * @param {String} verificationPayload.email - the email
+	 * @param {String} verificationPayload.token - the verification token
+	 *
+	 * @fulfil {String} - session token
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.auth.verifyEmail({
+	 * 	email: 'johndoe@gmail.com',
+	 * 	token: '5bb11d90eefb34a70318f06a43ef063f'
+	 * }).then(function(jwt) {
+	 * 	console.log(jwt);
+	 * });
+	 *
+	 */
+	async function verifyEmail(verificationPayload: {
+		email: string;
+		token: string;
+	}): Promise<string> {
+		const email = verificationPayload.email;
+		const verificationToken = verificationPayload.token;
+
+		const { body } = await request.send({
+			method: 'POST',
+			url: '/user/v1/verify-email',
+			body: {
+				verificationToken,
+				email,
+			},
+			baseUrl: apiUrl,
+			sendToken: false,
+		});
+
+		return body;
+	}
+
+	/**
+	 * @summary Re-send verification email to the user
+	 * @name requestVerificationEmail
+	 * @public
+	 * @function
+	 * @memberof balena.auth
+	 * @description This will only work if you used {@link balena.auth.login} to log in.
```

Comment:
```suggestion
```

Reply:
Any reason we want to remove this? This would be only used after `login` and we need the following to work as well: https://github.com/balena-io/balena-sdk/pull/1077/files#diff-cd41b427b38a554f9a1a2d2e848e41ddc39bc7d68dba54c1398d7384d6aaf4f7R543

Reply:
I mean that I expect this to work even if you have logged in with an named user API key, which is possible for pre-existing users.
Not a big deal though I guess.